pkg/resources/metrics.go

```diff
@@ -463,6 +463,70 @@ func CreateRedisMemoryUsageAlerts(ctx context.Context, client k8sclient.Client,
 	return nil
 }
 
+// CreateRedisCpuUsageHighAlerts creates a PrometheusRule alerts to watch for High Cpu usage
+// of a Redis cache
+func CreateRedisCpuUsageAlerts(ctx context.Context, client k8sclient.Client, inst *v1alpha1.RHMI, cr *crov1.Redis) error {
+	if strings.ToLower(inst.Spec.UseClusterStorage) == "true" {
+		logrus.Info("skipping redis memory usage high alert creation, useClusterStorage is true")
+		return nil
+	}
+
+	alertName := "RedisCpuUsageHigh"
+	ruleName := "redis-cpu-usage-high"
+	alertDescription := "Redis Cpu for instance {{ $labels.instanceID }} is 90 percent or higher for the last hour. Redis Custom Resource: {{ $labels.resourceID }} in namespace {{ $labels.namespace }} for the product: {{ $labels.productName }}"
+	labels := map[string]string{
+		"severity": "critical",
+	}
+
+	alertExp := intstr.FromString(fmt.Sprintf("cro_redis_engine_cpu_utilization_average > %s", alertPercentage))
+
+	_, err := reconcilePrometheusRule(ctx, client, ruleName, cr.Namespace, alertName, alertDescription, sopUrlRedisCpuUsageHigh, alertFor60Mins, alertExp, labels)
+	if err != nil {
+		return err
+	}
+
+	// job to check time that the operator metrics are exposed
+	job := "cloud-resource-operator-metrics"
+
+	alertName = "RedisCpuUsageMaxIn4Hours"
+	ruleName = "redis-cpu-usage-will-max-in-4-hours"
+	alertDescription = "Redis Memory Usage is predicted to max with in four hours for instance {{ $labels.instanceID }}. Redis Custom Resource: {{ $labels.resourceID }} in namespace {{ $labels.namespace }} for the product: {{ $labels.productName }}"
+	labels = map[string]string{
+		"severity": "critical",
+	}
+	// building a predict_linear query using 1 hour of data points to predict a 4 hour projection, and checking if it is less than or equal 0
+	//    * [1h] - one hour data points
+	//    * , 4 * 3600 - multiplying data points by 4 hours
+	// and matching by label `job` if the current time is greater than 1 hour of the process start time for the cloud resource operator metrics.
+	//    * on(job) - matching queries by label job across both metrics
+	alertExp = intstr.FromString(fmt.Sprintf("predict_linear(cro_redis_engine_cpu_utilization_average{job='%s'}[1h], 5 * 3600) >= 100 and on(job) (time() - process_start_time_seconds{job='%s'}) / 3600 > 1", job, job))
```

Comment:
Just thinking about these prediction alerts on CPU utilization. 

`predict_linear` seems to be used mostly for resource exhaustion (will run out of storage/memory). I am thinking would this lead to noisy alerts when trying to predict based on a counter for CPU utilization :thinking: 

Looking at [AWS docs](https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/CacheMetrics.WhichShouldIMonitor.html) should we be alerting based on a threshold (`RedisCpuUsageHigh`) only :thinking: 

What do you think?

Reply:
I'm not invested in predict_linear this pr is block till https://github.com/integr8ly/integreatly-operator/pull/1108 is merged then will refactor, redis is memory and network bound likely the memory will trigger well before we ever see issues with cpu.

Reply:
For the cpu usage for postgres we just added one alert for over 90% warning for 5 minutes. Seeing as it dropped immediately after any loading I figured we'd just need that one and no predict_linear ones.

Reply:
@laurafitzgerald :thinking: , 5 min might be a bit low, I went with 1 hour which might be too long , think think the sweet spot is some where in between. might go with 15min